sw/device/silicon_creator/lib/crypto/tests/README.md

```diff
@@ -0,0 +1,63 @@
+# Crypto Library Tests
+
+This folder contains test data and infrastructure for the OpenTitan
+cryptographic library. Tests come from a variety of sources, including:
+- Our own hard-coded tests and regression tests
+- Coming soon: [wycheproof](https://github.com/google/wycheproof)
+- Coming soon: random tests!
+- Coming soon: FIPS tests!
+
+## Setup
+
+Each algorithm has a C test file (ending in `_functest.c`) which reads test
+data from a header file (ending in `_testvectors.h`) and runs a set of tests on
+each test vector. By default, the header file contains a small set of
+hard-coded test vectors. If all you want to do is sanity-check, then simply run
+`ninja` and run the `_functest` target (e.g.
+`sw_silicon_creator_lib_crypto_rsa_3072_verify_functest`) on your desired
+platform.
+
+However, if you want to run a set of test vectors from an external source, a
+custom set of tests, or random tests, you'll need to populate the
+`*_testvectors.h` header file with the tests you want to run. There is a Python
```

Comment:
It may interest you to learn that the C preprocessor will actually inject defines into `#include`s. This works:

```c
#ifndef OH_NO
#define OH_NO "some_default_file.h"
#endif // OH_NO
#include OH_NO
```
This could be used, for example, to allow re-using `_functest.c` with different headers, allowing you to have separate static targets, by doing something like `-DRSA_TEST_VECTORS=my_cool_test_vectors.h` as a clang argument.

@cfrantz would you condone evil such as this?

Reply:
I think this would be useful if we prefer having different test "suites". Another alternative could be to slightly modify the python script to accept multiple hjson files. We can keep those hjson files separate but generate a single header that includes all the test vectors.